cluster/addons/dns/coredns/coredns.yaml.base

```diff
@@ -73,7 +73,9 @@ data:
             ttl 30
         }
         prometheus :9153
-        forward . /etc/resolv.conf
+        forward . /etc/resolv.conf {
+            max_concurrent 1000
```

Comment:
@prameshj 

We've had people run into this # of concurrents as an issue for DNS masq.
Is there a link or reference (embed as a comment) as to how this value was determined?

Reply:
Here was my reasoning: https://github.com/coredns/deployment/pull/233#issue-438151578

Reply:
My understanding was also that since this only applies to forwarded queries, cached responses and "cluster.local" queries would continue to be handled. Thanks for the explanation above. Good to know that the client will get a SERVFAIL rather than timeout after no response.

I am curious, how the memory overhead of 2 MB was measured?Any test that queried 1000 unique external names in parallel? From https://golang.org/doc/go1.2#stack_size, if a single goroutine took up 4KB, this could use upto 4MB?

Reply:
> My understanding was also that since this only applies to forwarded queries, cached responses and "cluster.local" queries would continue to be handled.

Yes, that's correct.

> From https://golang.org/doc/go1.2#stack_size, if a single goroutine took up 4KB, this could use upto 4MB?

I was basing it on 2k per go-routine https://golang.org/doc/go1.4#runtime.  Nonetheless, 2MB is admittedly an arbitrary number.  The default memory limit per CoreDNS Pod is 170Mi, so there is _a lot_ of headroom before an OOM happens.

If we are not comfortable picking number at this time, we can just leave the it out.  When people see OOMs due to query spikes, they can add this manually.

Reply:
Based on past issues that have been reported, most of the parallel queries are related to searchpath expansion and will have the "cluster.local" suffix. This will be handled by the  "kubernetes" plugin, and hence unaffected by the 1000 connection limit. It seems reasonable to add this new limit to prevent OOMs of coreDNS pods. @bowei what do you think?

Reply:
sgtm

/lgtm
/approve